## Analyzing CIA Factbook Data Using SQL

In this project, we'll work with data from the CIA World Factbook, a compendium of statistics about all of the countries on Earth. The Factbook contains demographic information like:

- population - The population as of 2015.
- population_growth - The annual population growth rate, as a percentage.
- area - The total land and water area.

Connecting the jupyter notebook with the data base.

In [1]:
%%capture
%load_ext sql
%sql sqlite:///factbook.db

'Connected: None@factbook.db'

In [2]:
%%sql
SELECT *
  FROM sqlite_master
 WHERE type='table';

Done.


[('table', 'sqlite_sequence', 'sqlite_sequence', 3, 'CREATE TABLE sqlite_sequence(name,seq)'),
 ('table', 'facts', 'facts', 47, 'CREATE TABLE "facts" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "code" varchar(255) NOT NULL, "name" varchar(255) NOT NULL, "area" integer, "a ... (4 characters truncated) ... land" integer, "area_water" integer, "population" integer, "population_growth" float, "birth_rate" float, "death_rate" float, "migration_rate" float)')]

In [3]:
%%sql
SELECT *
  FROM facts
 LIMIT 5;

Done.


[(1, 'af', 'Afghanistan', 652230, 652230, 0, 32564342, 2.32, 38.57, 13.89, 1.51),
 (2, 'al', 'Albania', 28748, 27398, 1350, 3029278, 0.3, 12.92, 6.58, 3.3),
 (3, 'ag', 'Algeria', 2381741, 2381741, 0, 39542166, 1.84, 23.67, 4.31, 0.92),
 (4, 'an', 'Andorra', 468, 468, 0, 85580, 0.12, 8.13, 6.96, 0.0),
 (5, 'ao', 'Angola', 1246700, 1246700, 0, 19625353, 2.78, 38.78, 11.49, 0.46)]

## Summary statistics

Writing a single query to return minimum population, maximum population, minimum population growth, maximum population growth.

In [4]:
%%sql
SELECT MIN(population) AS Minimum_population,
    MAX(population) AS Max_population,
    MIN(population_growth) AS Min_pop_growth,
    MAX(population_growth) AS Max_pop_growth
  FROM facts;

Done.


[(0, 7256490011, 0.0, 4.02)]

In [5]:
%%sql
SELECT name, population
    FROM facts
  WHERE population = (      
                    SELECT MIN(population)
                     FROM facts);

Done.


[('Antarctica', 0)]

Antartica has zero population. There is no permanent residents living in Antartica and the result also matches with  CIA Factbook data.

In [6]:
%%sql
SELECT name, population
    FROM facts
  WHERE population = (      
                    SELECT MAX(population)
                     FROM facts);

Done.


[('World', 7256490011)]

The table 'facts' contains a row named 'World' which has 7.2 billion population and is the total sum of the population of the whole world.

In [7]:
%%sql
SELECT name, population_growth
    FROM facts
  WHERE population_growth = (      
                    SELECT MAX(population_growth)
                     FROM facts);

Done.


[('South Sudan', 4.02)]

As seen from the result the country 'South Sudan' has the highest population growth rate of 4.02 % in the world.

In [8]:
%%sql
SELECT name, population_growth
    FROM facts
  WHERE population_growth = (      
                    SELECT MIN(population_growth)
                     FROM facts);

Done.


[('Holy See (Vatican City)', 0.0),
 ('Cocos (Keeling) Islands', 0.0),
 ('Greenland', 0.0),
 ('Pitcairn Islands', 0.0)]

### The row 'World' has maximum population as it is summation of the total population of the world. Thus we will recalculate the summary statistics excluding 'World'.

In [9]:
%%sql
SELECT MIN(population) AS Min_pop,
       MAX(population) AS Max_pop,
       MIN(population_growth) AS Min_pop_growth,
       MAX(population_growth) AS Max_pop_growth 
  FROM facts
 WHERE name <> 'World';

Done.


[(0, 1367485388, 0.0, 4.02)]

We have a country(s) with  nearly 1.37 billion population.

## Calculationg the Average population and Area.

We will calculate the average population of the world per country and average area per country excluding the 'World' row.


In [10]:
%%sql
SELECT AVG(population) AS Avg_Population,
  AVG(area) AS Avg_Area
    FROM facts
  WHERE name <> 'world';

Done.


[(62094928.32231405, 555093.546184739)]

Average population of the world is 62 million and average area is 555 thousand square kilometers.

## Finding densely populated countries.

We will identify countries that have:

- Above average values for population.
- Below average values for area.

In [ ]:
%%sql
SELECT name, population. area
    FROM facts
  WHERE population > (SELECT population
                        FROM facts)